caraml-store-registry/src/main/java/dev/caraml/store/sparkjob/JobService.java

```diff
@@ -556,4 +578,185 @@ private String getDeltaTableRef(String projectName, String featureTableName) {
         projectName,
         featureTableName);
   }
+
+  public BatchJobRecord newBatchJobRecordFromJob(Job job) {
+    if (job.getType().equals(JobType.STREAM_INGESTION_JOB)) {
+      throw new IllegalArgumentException(
+          "Cannot create BatchJobRecord from a stream ingestion job");
+    }
+    SparkApplication sparkApp =
+        this.sparkOperatorApi.getSparkApplication(namespace, job.getId()).orElseThrow();
+    String sparkAppManifestYaml = Yaml.dump(sparkApp);
+    if (job.getType().equals(JobType.BATCH_INGESTION_JOB)) {
+      String featureTableName = job.getBatchIngestion().getTableName();
+      // Retrieve feature table object from the database
+      FeatureTable table =
+          this.tableRepository
+              .findFeatureTableByNameAndProject_NameAndIsDeletedFalse(
+                  featureTableName, job.getProject())
+              .orElseThrow(
+                  () -> new FeatureTableNotFoundException(job.getProject(), featureTableName));
+      long jobStartTime = job.getStartTime().getSeconds();
+      Pair<java.sql.Timestamp, java.sql.Timestamp> startEndTimeParams =
+          BatchJobRecord.getStartEndTimeParamsFromSparkApplication(sparkApp);
+      String recordId =
+          sparkApp
+              .getMetadata()
+              .getLabels()
+              .getOrDefault(RECORD_JOB_LABEL, UUID.randomUUID().toString());
+      // check if SparkApplication is created from a scheduledSparkApplication
+      // if it is, set recordId to uuid-(sparkAppName suffix)
+      if (sparkApp
+          .getMetadata()
+          .getLabels()
+          .containsKey("sparkoperator.k8s.io/scheduled-app-name")) {
+        // get sparkapplication name suffix
```

Comment:
This step is performed because the `SparkApplication` created from the `ScheduledSparkapplication` will contain the same set of labels passed to the `ScheduledSparkapplication`. Since the record-id is passed only once, we need another way to differentiate Sparkapplications created from the same `ScheduledSparkapplication` in the db.

This is where the suffix in the Sparkapplication name is used, since all Sparkapplications created from a ScheduledSparkapplication has the format `caraml-65d8c6f9340adee8-1750762810975636102`, where `1750762810975636102` is unique.